component/src/main/java/io/siddhi/extension/io/file/listeners/FileCronExecutor.java

```diff
@@ -0,0 +1,212 @@
+                              Map<String, String> properties, String fileUri,
+                              FileSourceConfiguration fileSourceConfiguration) {
+        String actionAfterProcess = fileSourceConfiguration.getActionAfterProcess();
+        properties.put(Constants.URI, fileUri);
+        properties.put(Constants.ACK_TIME_OUT, "1000");
+        BinaryCarbonMessage carbonMessage = new BinaryCarbonMessage(ByteBuffer.wrap(
+                fileUri.getBytes(StandardCharsets.UTF_8)), true);
+        String moveAfterProcess = fileSourceConfiguration.getMoveAfterProcess();
+        try {
+            if (fileSourceConfiguration.getActionAfterProcess() != null) {
+                properties.put(Constants.URI, fileUri);
+                properties.put(Constants.ACTION, actionAfterProcess);
+                File file = new File(fileSourceConfiguration.getUri());
+                if (fileSourceConfiguration.getMoveAfterProcess() != null) {
+                    if (file.isFile()) {
+                        properties.put(Constants.DESTINATION, fileSourceConfiguration.getMoveAfterProcess());
+                    } else {
+                        String destination = constructPath(moveAfterProcess, getFileName(fileUri,
+                                fileSourceConfiguration.getProtocolForMoveAfterProcess()));
+                        if (destination != null) {
+                            properties.put(Constants.DESTINATION, destination);
+                        }
+                    }
+                }
+                vfsClientConnector.send(carbonMessage, vfsClientConnectorCallback, properties);
+                vfsClientConnectorCallback.waitTillDone(fileSourceConfiguration.getTimeout(), fileUri);
+            }
+        } catch (ClientConnectorException e) {
+            log.error(String.format("Failure occurred in vfs-client while reading the file '%s '.", fileUri), e);
+        } catch (InterruptedException e) {
+            log.error(String.format("Failed to get callback from vfs-client for file '%s '.", fileUri), e);
+        }
+    }
+
+    private String getFileName(String uri, String protocol) {
+        try {
+            URL url = new URL(String.format("%s%s%s", protocol, File.separator, uri));
+            return FilenameUtils.getName(url.getPath());
+        } catch (MalformedURLException e) {
+            log.error(String.format("Failed to extract file name from the uri '%s '.", uri), e);
+            return null;
+        }
+    }
+
+    private String constructPath(String baseUri, String fileName) {
+        if (baseUri != null && fileName != null) {
+            if (baseUri.endsWith(File.separator)) {
+                return String.format("%s%s", baseUri, fileName);
+            } else {
+                return String.format("%s%s%s", baseUri, File.separator, fileName);
+            }
+        } else {
+            return null;
+        }
+    }
```

Comment:
Shall we add these 2 methods to a util class and also use those util methods here[https://github.com/siddhi-io/siddhi-io-file/blob/master/component/src/main/java/io/siddhi/extension/io/file/listeners/FileSystemListener.java#L267](url) to avoid code duplications?

Reply:
fixed in 33a0d4ec677b82f844777943dec4b611303e759d